src/navigations/HomeBottomTabs.js

```diff
@@ -207,135 +211,116 @@ function HomeBottomTabs({navigation}) {
+            route.name === 'AnonymousChannelList' ? colors.anon_primary : colors.darkBlue
         })}>
         <Tab.Screen
-          name="Feed"
-          component={FeedScreen}
-          initialParams={{isBottomTab: true}}
-          options={{
-            activeTintColor: colors.holytosca,
-            tabBarIcon: renderTabLabelIcon('Feed')
-            // unmountOnBlur: true
-          }}
-        />
-        {/* <Tab.Screen
-          name="Feed"
-          component={WebsocketResearchScreen}
-          initialParams={{isBottomTab: true}}
-          options={{
-            activeTintColor: colors.holytosca,
-            tabBarIcon: renderTabLabelIcon('Feed')
-            // unmountOnBlur: true
-          }}
-        /> */}
-        {/* <Tab.Screen
-          name="ChannelList"
+          name="SignedChannelList"
           component={ChannelListScreen}
           initialParams={{isBottomTab: true}}
-          options={{
-            activeTintColor: colors.holytosca,
-            tabBarIcon: renderTabLabelIcon('ChannelList'),
-            tabBarBadge:
-              unReadMessage.total_unread_count + unReadMessage.unread_post > 0
-                ? unReadMessage.total_unread_count + unReadMessage.unread_post
-                : null
-          }}
-        /> */}
+          listeners={({route}) => saveLastMenu(route)}
+          options={{tabBarBadge: signedChannelUnreadCount > 0 ? signedChannelUnreadCount : null}}
+        />
         <Tab.Screen
-          name="ChannelList"
+          name="AnonymousChannelList"
           component={ChannelListScreenV2}
           initialParams={{isBottomTab: true}}
+          listeners={({route}) => saveLastMenu(route)}
           options={{
-            activeTintColor: colors.holytosca,
-            tabBarIcon: renderTabLabelIcon('ChannelList'),
-            tabBarBadge: totalUnreadCount > 0 ? totalUnreadCount : null
+            tabBarBadge: anonymousChannelUnreadCount > 0 ? anonymousChannelUnreadCount : null
           }}
         />
+        <Tab.Screen
+          name="Feed"
+          component={FeedScreen}
+          initialParams={{isBottomTab: true}}
+          listeners={({route}) => saveLastMenu(route)}
+        />
         <Tab.Screen
           name="News"
           component={NewsScreen}
           initialParams={{isBottomTab: true}}
-          options={{
-            activeTintColor: colors.holytosca,
-            tabBarIcon: renderTabLabelIcon('News')
-            // unmountOnBlur: true
-          }}
+          listeners={({route}) => saveLastMenu(route)}
         />
         <Tab.Screen
           name="Profile"
-          initialParams={{isBottomTab: true}}
           component={ProfileScreen}
-          options={{
-            activeTintColor: colors.holytosca,
-            tabBarIcon: renderTabLabelIcon('Profile')
-            // unmountOnBlur:true
-          }}
+          initialParams={{isBottomTab: true}}
+          listeners={({route}) => saveLastMenu(route)}
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

In the `Tab.Navigator` component, you're repeating similar code for each `Tab.Screen`. You could create a function that returns a `Tab.Screen` component to reduce repetition. This would make your code more DRY (Don't Repeat Yourself) and easier to maintain.

```javascript
const renderTabScreen = (name, component, badgeCount) => (
  <Tab.Screen
    name={name}
    component={component}
    initialParams={{isBottomTab: true}}
    listeners={({route}) => saveLastMenu(route)}
    options={{tabBarBadge: badgeCount > 0 ? badgeCount : null}}
  />
);

return (
  <View style={styles.container}>
    <Tab.Navigator
      initialRouteName={getInitialRouteName()}
      tabBarOptions={{
        inactiveTintColor: colors.gray1,
        safeAreaInsets: {top: 0, bottom: 0, left: 0, right: 0}
      }}
      screenOptions={({navigation: nav, route}) => ({
        tabBarLabel: () => menuIndicator(nav, route),
        tabBarIcon: ({focused, color}) => renderTabBarIcon(route, focused, color),
        tabBarActiveTintColor:
          route.name === 'AnonymousChannelList' ? colors.anon_primary : colors.darkBlue
      })}>
      {renderTabScreen("SignedChannelList", ChannelListScreen, signedChannelUnreadCount)}
      {renderTabScreen("AnonymousChannelList", ChannelListScreenV2, anonymousChannelUnreadCount)}
      {renderTabScreen("Feed", FeedScreen)}
      {renderTabScreen("News", NewsScreen)}
      {renderTabScreen("Profile", ProfileScreen)}
    </Tab.Navigator>
    <FirebaseConfig navigation={navigation} />
  </View>
);
```



<!-- This is an auto-generated comment by OSS CodeRabbit -->